toml/_dotty.py

```diff
@@ -103,6 +104,7 @@ def _get_or_create(self, item: dict, k: str, global_key: str) -> dict:
 
             item[k] = {}
 
+        del global_key, k, item
```

Comment:
Can't delete `item` nor `k` as we need to return `item[k]` (unless we first store them in another var and return that one instead)
```suggestion
        del global_key
```